src/ServiceInsight/ServiceControl/ServiceControlConnectionProvider.cs

```diff
@@ -1,20 +1,33 @@
 ﻿namespace ServiceInsight.ServiceControl
 {
     using Caliburn.Micro;
+    using ServiceInsight.Framework;
 
     public class ServiceControlConnectionProvider
     {
         IEventAggregator eventAggregator;
+		IRxServiceControl serviceControl;
 
-        public ServiceControlConnectionProvider(IEventAggregator eventAggregator)
+        public ServiceControlConnectionProvider(IEventAggregator eventAggregator, IRxServiceControl serviceControl)
         {
             this.eventAggregator = eventAggregator;
+			this.serviceControl = serviceControl;
         }
 
         public void ConnectTo(string url)
         {
             Url = url;
-            eventAggregator.PublishOnUIThread(new ServiceControlConnectionChanged());
+			
+			eventAggregator.PublishOnUIThread(new ServiceControlConnectionChanged());
```

Comment:
Hmm, yes, but theoretically the things that use the new RxServiceControl won't need to listen to that event because they should get notified something changed reactively. I'll move the call anyway because yes it should change, then notify of change.